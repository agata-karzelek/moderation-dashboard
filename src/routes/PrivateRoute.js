import React from "react";
import { Route, Redirect } from "react-router-dom";
import { isAuth } from "../services/localStorage";

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={(props) =>
      isAuth() ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: "/",
            state: { from: props.location },
          }}
        />
      )
    }
  ></Route>
);

export default PrivateRoute;
