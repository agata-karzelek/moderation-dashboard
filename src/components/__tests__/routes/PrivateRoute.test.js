import React from "react";
import { shallow } from "enzyme";
import PrivateRoute from "../../../routes/PrivateRoute";

describe("Test Cases for Private Route", () => {
  test("Snapshot", () => {
    const wrapper = shallow(<PrivateRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
