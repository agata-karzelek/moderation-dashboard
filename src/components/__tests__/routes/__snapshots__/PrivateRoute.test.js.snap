// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for Private Route Snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
